analyzers/src/SonarAnalyzer.CSharp/Helpers/CSharpSyntaxHelper.cs

```diff
@@ -167,23 +167,15 @@ public static bool IsStringEmpty(this ExpressionSyntax expression, SemanticModel
             .Any(symbolPredicate);
     }
 
-    public static SyntaxToken? GetIdentifierOrDefault(this BaseMethodDeclarationSyntax methodDeclaration)
-    {
-        switch (methodDeclaration?.Kind())
+    public static SyntaxToken? GetIdentifierOrDefault(this BaseMethodDeclarationSyntax methodDeclaration) =>
+        methodDeclaration?.Kind() switch
         {
-            case SyntaxKind.ConstructorDeclaration:
-                return ((ConstructorDeclarationSyntax)methodDeclaration).Identifier;
-
-            case SyntaxKind.DestructorDeclaration:
-                return ((DestructorDeclarationSyntax)methodDeclaration).Identifier;
-
-            case SyntaxKind.MethodDeclaration:
-                return ((MethodDeclarationSyntax)methodDeclaration).Identifier;
-
-            default:
-                return null;
-        }
-    }
+            SyntaxKind.ConstructorDeclaration => ((ConstructorDeclarationSyntax)methodDeclaration)?.Identifier,
+            SyntaxKind.DestructorDeclaration => ((DestructorDeclarationSyntax)methodDeclaration)?.Identifier,
+            SyntaxKind.MethodDeclaration => ((MethodDeclarationSyntax)methodDeclaration)?.Identifier,
+            _ when LocalFunctionStatementSyntaxWrapper.IsInstance(methodDeclaration) => ((LocalFunctionStatementSyntaxWrapper)methodDeclaration).Identifier,
```

Comment:
We cannot cast to LocalFunctionStatementSyntax because we don't have it, so it would be something like:
`SyntaxKindEx.LocalFunctionStatement => ((LocalFunctionStatementSyntaxWrapper)methodDeclaration).Identifier,`
I'm not a fan of it, I prefer using `LocalFunctionStatementSyntaxWrapper.IsInstance` to reduce the assumptions.

Reply:
But it's the same assumption as in the three cases above.